### DISCUSSION TITLE
Need to convert an image (jpg or png) to an 8bit Bitmap with only 1 channel (256 greyscale)

### DISCUSSION THREAD

**Discussion by v11** [ImageMagick/ImageMagick#5077] [Help] [answered]
Version:
ImageMagick 7.1.0-30

I tried different commands to achieve this:
`magick image.png -colorspace Gray BMP3:image.bmp`

The greyscale conversion works, but the export is always a 24bit RGB image (8bit per channel). So, i have 3-times the same data in my bytes.

In photoshop i can set the mode to grayscale and then export as 8bit bitmap.

<img width="830" alt="bitmap" src="https://user-images.githubusercontent.com/18447866/164971157-6f6152c5-9a58-4e4b-a62d-86cccebd9020.png">

How can I achieve this with ImageMagick?

**Comment by snibgo** [comment, marked as answer]
Your image may have more than 256 shades of gray, so reduce the image to 256 colours:
```
magick image.png -colorspace Gray -colors 256 	BMP3:image.bmp
```

**Comment by v11** [comment]
Thank you! 
 
When i test this locally with ImageMagick 7 it works. In my Cloud Function in Firebase (which is running 6.x) it "sometimes" works. But most of the time, the output is RGB (8bit per channel). Very strange.

Do you have any idea if -colors should work fine in 6.x?

**Comment by snibgo** [comment]
It works fine for me with v6.9.9-50 (on Windows 8.1). If your version is older than that, I suggest you upgrade. Preferably to v7.

**Comment by v11** [comment]
Unfortunately i can not update. Google Firebase is using v6.

But i figured out, it works with JPGs but not with PNGs. Do you have any idea to make this also compatible with PNG?

**Reply by v11** [reply]
OK, its seems that -alpha off did the trick.

**Comment by snibgo** [comment]
> ... it works with JPGs but not with PNGs.

JPGs have only 8 bits/channel, so a grayscale JPG file can have only 256 shades of gray. PNGs can have 16 bits/channel.

And JPGs cannot record alpha.
